### PR TITLE
Nav: add horizontal props to support rendering Nav horizontally.

### PR DESCRIPTION
**Pull request by inateeg**
#### Pull request checklist

- [ ] Addresses an existing issue: #0000
- [x] Include a change request file if publishing 
- [x] New feature, bugfix, or enhancement
  - [x] Includes tests
- [x] Documentation update

#### Description of changes

Support Nav component horizontal rendering

#### Focus areas to test

Nav

### CODE REVIEW

File: common/changes/inateeg-1201popup-elit_2017-01-13-00-11.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "office-ui-fabric-react",
+      "comment": "Support Nav horizontal orentation rendering",
```
Comment (dzearing) [inline common/changes/inateeg-1201popup-elit_2017-01-13-00-11.json]: This language needs to be in a better format; this goes into the public release notes, so please use this format:

Component: Full sentence explaining the new fix/change/feature.

Example:

Nav: Now adding the ability to render the Nav horizontally, if the `horizontal` attribute is provided.

File: packages/office-ui-fabric-react/src/components/Nav/Nav.Props.ts [hunk 2]
```diff
@@ -28,7 +28,7 @@ export interface INavProps {
   onLinkClick?: (ev?: React.MouseEvent<HTMLElement>, item?: INavLink) => void;
 
   /**
-   * Indicates whether the navigation component renders on top of other content in the UI
+   * Indicates whether the navigation component renders on top of other content in the UI (T-Narrow case)
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Nav/Nav.Props.ts]: not sure what tnarrow is.

File: packages/office-ui-fabric-react/src/components/Nav/Nav.tsx [hunk 3]
```diff
@@ -36,6 +36,7 @@ export class Nav extends React.Component<INavProps, INavState> implements INav {
   };
 
   private _hasExpandButton: boolean;
+  private _horizontal: boolean;
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Nav/Nav.tsx]: why do you need to cache this.props.horizontal in a member variable?

Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Nav/Nav.tsx]: Also if the caller were to provide a true/false toggle, this wouldn't work.

File: packages/office-ui-fabric-react/src/components/Nav/Nav.scss [hunk 4]
```diff
@@ -96,6 +96,11 @@ $linkRightPadding: 20px;
   vertical-align: middle;
 }
 
+.ms-Nav-linkHorizontal {
+  float: left;
```
Comment (cschlechty) [inline packages/office-ui-fabric-react/src/components/Nav/Nav.scss]: localize

File: packages/office-ui-fabric-react/src/components/Nav/Nav.Props.ts [hunk 5]
```diff
@@ -43,6 +43,11 @@ export interface INavProps {
   selectedKey?: string;
 
   /**
+   * (Optional) The nav links horizontal orentation flag.
```
Comment (mdahamiwal) [inline packages/office-ui-fabric-react/src/components/Nav/Nav.Props.ts]: `orientation` typo

File: packages/office-ui-fabric-react/src/components/Nav/Nav.scss [hunk 6]
```diff
@@ -96,6 +96,11 @@ $linkRightPadding: 20px;
   vertical-align: middle;
 }
 
+.ms-Nav-linkHorizontal {
+  @include float(left);
+  display: inline;
```
Comment (mdahamiwal) [inline packages/office-ui-fabric-react/src/components/Nav/Nav.scss]: do we need display: inline if we are floating the elements?

Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Nav/Nav.scss]: We shouldn't. Personally not a fan of float left if you can avoid it; display: inline-block is a bit more predictable.

Comment (mdahamiwal) [inline packages/office-ui-fabric-react/src/components/Nav/Nav.scss]: Just curious, if `float:left / float:right` is the only requirement, Can't we just provide `className` on `<ul>` and `<li>`s and let consumer take care of overriding this style whichever way they want. Do we really need an additional option for one style? 
👍 for adding an example that demonstrate horizontal styling.

Comment (inateeg) [inline packages/office-ui-fabric-react/src/components/Nav/Nav.scss]: in the debug session, seems display: inline alone is not enough.  As for allow consumer to override the class, we can if need is there.  Currently with @include float(left) it should take care of local LTR or RTL issue.

Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Nav/Nav.scss]: did you try inline-block? You shouldn't need float, and normal inline-block will do stacking like you'd expect. You may need to overflow clip things.

**Comment by inateeg** [comment]
I tried inline-block.

**Comment by inateeg** [comment]
Inline-block no difference.  Seems float: left is key aspect need to render horizontal

**Comment by dzearing** [comment]
Per discussion, let's hold off checking this in here.

**Comment by inateeg** [comment]
yes

**Pull request closed without merging**